sgRNA_design/sgRNA_finder.py

```diff
@@ -0,0 +1,98 @@
+from collections import defaultdict
+
+# Function to determine if a potential PAM site meets the specified range and 
+# efficiency values. pamLine should be one line of csv with format:
+# Start Pos,Stop Pos, sgRNA sequence, PAM sequence (NGG), strand, MIT score
+def pam_In_Range(pamLine,geneRange,effCutOff=0):
+
+	# Parse and split csv line into list of string
+	words = pamLine.strip("\n").strip(",").split(',')
+	# Calculate the end position for PAM sequence based on desired coverage
+	endRange = (geneRange[1] - geneRange[0])*geneRange[2] + geneRange[0]
+
+	# Can be called directly in if and return statements, but makes code
+	# difficult to read
+	pamSta = int(words[0]) # Start Position
+	pamStp = int(words[1]) # Stop Position
+	pamSeq = words[2]      # sgRNA sequence
+	pamNGG = words[3]      # PAM sequence
+	pamStr = int(words[4]) # plus or minus strand
+	pamEff = float(words[5]) # MIT efficiency score
+
+	# determine if the PAM falls within the desired start and end sequence, if
+	# it meets the desired efficiency cut off (set to 0 by default) and whether
+	# it has masked characters in the sequence. Returns any PAMs that pass as 
+	# a list of values. Returns empty string if it fails because None value
+	# causes error in generating dictionary from returned values
+	if pamSta > geneRange[0] and pamStp < endRange and 'N' not in pamSeq and \
+	pamEff > effCutOff:
```

Comment:
I'd suggest all(cond1,
                       cond2, etc) 
instead of repeated and's